controllers/directivebreakdown_controller.go

```diff
@@ -115,17 +116,30 @@ func (r *DirectiveBreakdownReconciler) Reconcile(ctx context.Context, req ctrl.R
 			return ctrl.Result{}, nil
 		}
 
-		// TODO: Teardown directiveBreakdown
-		/*
-			if err := r.teardownDirectiveBreakdown(ctx, dbd); err != nil {
-				return ctrl.Result{}, err
-			}
+		// Delete all  children that are owned by this DirectiveBreakdown.
+		deleteStatus, err := dwsv1alpha1.DeleteChildren(ctx, r.Client, r.ChildObjects, dbd)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
 
-			controllerutil.RemoveFinalizer(dbd, finalizerDirectiveBreakdown)
-			if err := r.Update(ctx, dbd); err != nil {
-				return ctrl.Result{}, err
-			}
-		*/
+		if deleteStatus == dwsv1alpha1.DeleteRetry {
+			return ctrl.Result{}, nil
+		}
+
+		controllerutil.RemoveFinalizer(dbd, finalizerDirectiveBreakdown)
+		if err := r.Update(ctx, dbd); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		return ctrl.Result{}, nil
+	}
+
+	// Add the finalizer if it doesn't exist
+	if !controllerutil.ContainsFinalizer(dbd, finalizerDirectiveBreakdown) {
+		controllerutil.AddFinalizer(dbd, finalizerDirectiveBreakdown)
+		if err := r.Update(ctx, dbd); err != nil {
+			return ctrl.Result{Requeue: true}, nil
```

Comment:
I've never seen an error other than IsConflict here so it may be unncessary, but elsewhere we return with err if `!apierrors.IsConflict(err)`

Reply:
Okay, I'll change it